core/vcpu.c

```diff
@@ -2715,7 +2715,43 @@ static void handle_cpuid_virtual(struct vcpu_t *vcpu, uint32_t a, uint32_t c)
         case 3:                         // Reserved
         case 4: {                       // Deterministic Cache Parameters
             // [31:26] cores per package - 1
-            state->_eax = state->_ebx = state->_ecx = state->_edx = 0;
+            // Using QEMU's values
```

Comment:
Is there any specific reason using these hard-coded cache values for CPUID 4 handling? Do you consider using host cache parameters instead of hard-coded?

Reply:
These are values returned by qemu for most processors, so these are tested and proven to work.
Host parameters work too, but different values may lead to different execution paths in the guest and make possible debugging harder.